### PR TITLE
Clone accessories

### PR DESCRIPTION
**Pull request by akemidx**
Ability to clone accesories has been added.  While viewing an accessory in detail, the dropdown will include the option to clone that accessory.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] I have read the Contributing documentation available here: https://snipe-it.readme.io/docs/contributing-overview
- [x] I have formatted this PR according to the project guidelines: https://snipe-it.readme.io/docs/contributing-overview#pull-request-guidelines
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by akemidx** [comment]
Screen Shot 2023-03-15 at 5 02 01 PM

File: app/Http/Transformers/AccessoriesTransformer.php [hunk 1]
```diff
@@ -51,6 +51,8 @@ public function transformAccessory(Accessory $accessory)
             'checkin' =>  false,
             'update' => Gate::allows('update', Accessory::class),
             'delete' => Gate::allows('delete', Accessory::class),
+            'clone' => (Gate::allows('create', Accessory::class) && ($accessory->created_at == 'datetime')),
```
Comment (snipe) [inline app/Http/Transformers/AccessoriesTransformer.php]: Do we need the `&& ($accessory->created_at == 'datetime')`? here? Not sure what it does.

Comment (snipe) [inline app/Http/Transformers/AccessoriesTransformer.php:54]: Do we need the `&& ($accessory->created_at == 'datetime')`? here? Not sure what it does.

**Comment by clpStephen** [comment]
Thanks!!

File: app/Http/Controllers/Accessories/AccessoriesController.php [hunk 2]
```diff
@@ -115,6 +115,33 @@ public function edit($accessoryId = null)
...
+     */
+    public function getClone($accessoryId = null)
+    {
+
+        $this->authorize('create', Accesory::class);
+
+        // Check if the asset exists
+        if (is_null($accessory_to_clone = Accessory::find($accessoryId))) {
+            // Redirect to the asset management page
+            return redirect()->route('accessory.index')->with('error', trans('admin/accessories/message.does_not_exist'));
```
Comment (snipe) [inline app/Http/Controllers/Accessories/AccessoriesController.php:134]: Routes are usually plural, so this should have been `accessories.index`, versus `accessory.index`. Fixed in #13439

File: app/Http/Controllers/Accessories/AccessoriesController.php [hunk 3]
```diff
@@ -115,6 +115,33 @@ public function edit($accessoryId = null)
...
+     *
+     * @author [J. Vinsmoke]
+     * @param int $accessoryId
+     * @since [v6.0]
+     * @return View
+     */
+    public function getClone($accessoryId = null)
+    {
+
+        $this->authorize('create', Accesory::class);
```
Comment (snipe) [inline app/Http/Controllers/Accessories/AccessoriesController.php:129]: Also a typo here - I just fixed it tho

**Pull request merged**